### PR TITLE
[feature] apiv2 absolute URL convenience functions

### PR DESCRIPTION
**Pull request by abought**
References brianjgeiger/osf.io/issues/90 . 

So far this is only the python function; PR created for tracking purposes. The JS URL and request helpers will be added next.

### CODE REVIEW

File: website/util/__init__.py [hunk 1]
```diff
@@ -60,6 +56,28 @@ def api_url_for(view_name, _absolute=False, _offload=False, _xml=False, *args, *
     return url
 
 
+def api_v2_url_for(path_str, params=None, base_route=settings.API_DOMAIN, **kwargs):
```
Comment (brianjgeiger) [inline website/util/__init__.py]: Let's call this api_v2_url, since it's not doing lookup.

File: tests/test_utils.py [hunk 2]
```diff
@@ -59,6 +59,20 @@ def test_api_url_for(self):
         with self.app.test_request_context():
             assert api_url_for('dummy_view', pid='123') == '/api/v1/123/'
 
+    def test_api_v2_url_for_with_port(self):
```
Comment (brianjgeiger) [inline tests/test_utils.py]: :+1:

File: tests/test_utils.py [hunk 3]
```diff
@@ -59,6 +59,20 @@ def test_api_url_for(self):
...
+        """Handles URLs with port correctly"""
+        full_url = api_v2_url_for('/nodes/abcd3/contributors/',
+                                  base_route='http://localhost:8000/')
+        assert full_url == "http://localhost:8000/nodes/abcd3/contributors/"
```
Comment (brianjgeiger) [inline tests/test_utils.py]: I believe current best practice is to use assert_equal(item1, item2)

**Comment by brianjgeiger** [comment]
First pass done. :penguin:

**Comment by abought** [comment]
Wasn't sure whether this was still slated for 0.38, so I updated this PR from `brianjgeiger/osf.io/api-v2-drf` branch.

New feature: adds a helper function to correctly generate API urls (including base server) from JS.

Internally, the base URL used in the JS function is actually calculated by the python version, which should help preserve some level of consistency. Usage examples are provided.

JS CORS request helpers will be implemented in a separate pull request. Feedback, review, and advice on suggested branch target for this PR welcome when ready.

**Comment by lbanner** [comment]
Good @abought ... I'll let @brianjgeiger and @sloria review and note if / when ready. TY

**Comment by brianjgeiger** [comment]
It looks like this has pulled from my branch rather than just being 0.38. Go ahead and submit to my repo and close this one (or redo just pulling from 0.38)

**Pull request closed without merging**